redis/src/main/scala/zio/redis/api/Strings.scala

```diff
@@ -254,6 +254,18 @@ trait Strings {
   final def setRange(key: String, offset: Long, value: String): ZIO[RedisExecutor, RedisError, Long] =
     SetRange.run((key, offset, value))
 
+  /**
+   * Implements complex algorithms that operate on strings. Right now the only algorithm implemented is
+   * the longest common subsequence.
+   *
+   * @param key Key of the string to get the length of
+   * @return Returns the length of the string
+   */
+  final def strAlgo(command: StrAlgoCommand): ZIO[RedisExecutor, RedisError, LongestCommonSubsequenceResult] =
```

Comment:
I recommend that you create `strAlgoLcs` function that will receive actual arguments. Also, when new options are added, the output will probably be of a different type, so it will be easier to create new functions.

Reply:
Got it, wasn't sure what kind of API we would be best. So instead we should use the following:

`final def strAlgoLcs(lcsType: LcsType,
      input1: String,
      input2: String,
      len: Option[QueryType] = None,
      minMatchLen: Option[MinMatchLen] = None,
      withMatchLen: Option[WithMatchLen] = None
): ZIO[RedisExecutor, RedisError, LongestCommonSubsequenceResult]`

I can also remove `case class LongestCommonSubsequence()` and simplify its inputs accordingly. Does this seem reasonable?